packages/object-utilities/src/js/__protected.js

```diff
@@ -0,0 +1,44 @@
+/**
```

Comment:
I want to understand why we need the `protected` function? what is the problem we are trying to solve here? I understand what is happening but why?